CMakeLists.txt

```diff
@@ -84,6 +86,11 @@ if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
   endif()
 endif()
 
+if (ENABLE_NATIVE_OPTIMIZATION)
+  set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -march=native -mtune=native")
+  set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native -mtune=native")
```

Comment:
```suggestion
  add_compile_options(-march=native -mtune=native)
```

Reply:
Are these flags valid to nvcc? (at least no errors)

Reply:
Maybe use this one
```cmake
add_compile_options(
    "$<$<COMPILE_LANGUAGE:C>:-march=native -mtune=native>"
    "$<$<COMPILE_LANGUAGE:CXX>:-march=native -mtune=native>"
)
```

Reply:
> Are these flags valid to nvcc? (at least no errors)

I think no errors means it's valid for nvcc. It will throw an error if it can not understand the flags.

Reply:
> Maybe use this one

I think simply `add_compile_options` is OK. Using generator expressions can be confusing. There's no need to indicate the language, since we only use C++.

Reply:
Ok, I manually tested `nvcc` and got the following errors.
```sh
nvcc -march=native -mtune=native
```

```
nvcc fatal   : Unknown option '-march=native'
```

Reply:
> since we only use C++.

Please see line 154...

https://github.com/deepmodeling/abacus-develop/blob/fb8a1544af0560396ffc77c22c91569afff38b48/CMakeLists.txt#L154

Reply:
Haha, it seems that your original commit is the best solution.